src/group_theory/submonoid.lean

```diff
@@ -151,6 +159,10 @@ assume a ha, by induction ha; simp [h _, *, is_submonoid.one_mem, is_submonoid.m
 theorem closure_mono {s t : set α} (h : s ⊆ t) : closure s ⊆ closure t :=
 closure_subset $ set.subset.trans h subset_closure
 
+theorem closure_singleton {x : α} : closure ({x} : set α) = powers x :=
```

Comment:
Wouldn't it make sense to make this the definition of `powers`? And turn the current definition into this theorem?

Reply:
I don't feel strongly about this either way, but I would guess that `∈ powers x` is probably proved directly from the current definition quite a few times in mathlib, and all of these would now require a lemma application.

Reply:
Chris has just been convincing me that the mathlib definition of `powers` is the best idea in the long run. Even though inductive props are the best, the closure of a singleton has a recursor which is too complicated because the recursor for closure in general is more complicated. So you would write your own recursor, saying that 1 is in, and if t is in then tx is in, and oh look you just wrote the recursor for nat, which we had already from the current definition of powers. Constructing an element like x^2 is also a pain using the inductive prop, whereas using the nat definition you can just say "use 2".